### ISSUE TITLE
I have added the frequency-domain Res FFT-ReLU Block to NAFBlock(Concat) according to the figures in your paper, but the results are not good. Would you mind checking whether I did it correctly?

### ISSUE THREAD

**Issue by bestzdy** [DeepMed-Lab/DeepRFT-AAAI2023#6]
if self.usefft:
            self.fftconv1 = nn.Sequential(
                nn.Conv2d(in_channels=c*2, out_channels=c*2, kernel_size=1, padding=0, stride=1, groups=1, bias=True),
                nn.GELU(),
                nn.Conv2d(in_channels=c*2, out_channels=c*2, kernel_size=1, padding=0, stride=1, groups=1, bias=True)
            )
            self.fftconv2 = nn.Sequential(
                nn.Conv2d(in_channels=c*2, out_channels=c*2, kernel_size=1, padding=0, stride=1, groups=1, bias=True),
                nn.GELU(),
                nn.Conv2d(in_channels=c*2, out_channels=c*2, kernel_size=1, padding=0, stride=1, groups=1, bias=True)
            )
def forward(self, inp):
        x = inp

        x1 = self.norm1(x)
        x = self.conv1(x1)
        x = self.conv2(x)
        x = self.sg(x)
        x = x * self.sca(x)
        x = self.conv3(x)

        x = self.dropout1(x)

        y = inp + x * self.beta

        if self.usefft:
            xf = torch.fft.rfft2(x1, norm="backward")# <---here，前半部分
            xf_imag = xf.imag
            xf_real = xf.real
            yf = torch.cat([xf_real, xf_imag], dim=1)
            ffty = self.fftconv1(yf)
            y_real, y_imag = torch.chunk(ffty, 2, dim=1)
            ffty = torch.complex(y_real, y_imag)
            fft_y = torch.fft.irfft2(ffty, norm="backward")
            y = y + fft_y * self.beta  #（y = inp + （x+fft_y）*self.beta）

        x2 = self.norm2(y)
        x = self.conv4(x2)
        x = self.sg(x)
        x = self.conv5(x)

        x = self.dropout2(x)
        out = y + x * self.gamma

        if self.usefft:
            xf = torch.fft.rfft2(x2, norm="backward")# <--here，后半部分
            xf_imag = xf.imag
            xf_real = xf.real
            yf = torch.cat([xf_real, xf_imag], dim=1)
            ffty = self.fftconv2(yf)
            y_real, y_imag = torch.chunk(ffty, 2, dim=1)
            ffty = torch.complex(y_real, y_imag)
            fft_y = torch.fft.irfft2(ffty, norm="backward")
            out = out + fft_y * self.gamma #（out = y+ （x+fft_y）*self.gamma）

        return  out

**Comment by INVOKERer** [comment]
Hi, maybe you miss the activation(GELU or ReLU) in the Fourier domain. Also, you can get the FNAFBlock in our [AdaRevD](https://github.com/INVOKERer/AdaRevD).
